cmd/sink/expr/cel.go

```diff
@@ -0,0 +1,211 @@
+		resourcesBytes, err := os.ReadFile(filePath)
+		if err != nil {
+			errList = append(errList, err)
+			continue
+		}
+		resources := []v1alpha1.KubeArchiveConfigResource{}
+		err = yaml.Unmarshal(resourcesBytes, &resources)
+		if err != nil {
+			errList = append(errList, err)
+			continue
+		}
+		for _, resource := range resources {
+			namespaceGvk := NamespaceGroupVersionKind(namespace, resource)
+			archiveExpr, err := CreateCelExpr(resource.ArchiveWhen)
+			if err != nil {
+				errList = append(errList, err)
+			} else {
+				filters.archive[namespaceGvk] = *archiveExpr
+			}
+			deleteExpr, err := CreateCelExpr(resource.DeleteWhen)
+			if err != nil {
+				errList = append(errList, err)
+			} else {
+				filters.delete[namespaceGvk] = *deleteExpr
+			}
+		}
+	}
+
+	return filters, errors.Join(errList...)
+
+}
+
+// Returns whether obj needs to be archived. Obj needs to be archived if any of the cel programs in archive return true
+// or if obj needs to be deleted. If obj is nil, it returns false
+func (f *Filters) MustArchive(ctx context.Context, obj *unstructured.Unstructured) bool {
+	if obj == nil {
+		return false
+	}
+	mustArchive := f.MustDelete(ctx, obj)
+	if mustArchive {
+		return mustArchive
+	}
+	globalGvk := GlobalGroupVersionKind(obj)
+	namespaceGvk := NamespaceGroupVersionKindFromObject(obj)
+	program, exists := f.archive[globalGvk]
+	if exists {
+		mustArchive = executeCel(ctx, program, obj)
+	}
+	if mustArchive {
+		return mustArchive
+	}
+	program, exists = f.archive[namespaceGvk]
+	if exists {
+		mustArchive = executeCel(ctx, program, obj)
+	}
+	return mustArchive
+}
+
+// Returns whether obj needs to be deleted. Obj needs to be deleted if the cel program in delete that matches the
+// NamespaceGroupVersionKind of obj or the cel program for the GlobalGroupVersionKind in delete for obj returns true. If
+// obj is nil, it returns false
+func (f *Filters) MustDelete(ctx context.Context, obj *unstructured.Unstructured) bool {
+	if obj == nil {
+		return false
+	}
+	mustDelete := false
+	globalGvk := GlobalGroupVersionKind(obj)
+	namespaceGvk := NamespaceGroupVersionKindFromObject(obj)
+	program, exists := f.delete[globalGvk]
+	if exists {
+		mustDelete = executeCel(ctx, program, obj)
+	}
+	if mustDelete {
+		return mustDelete
+	}
+	program, exists = f.delete[namespaceGvk]
+	if exists {
+		mustDelete = executeCel(ctx, program, obj)
+	}
+	return mustDelete
+}
+
+// Returns a string that is the form <globalKey>:GroupVersionKind
+func GlobalGroupVersionKind(obj *unstructured.Unstructured) string {
+	if obj == nil {
+		return ":"
+	}
+	return fmt.Sprintf("%s:%s", globalKey, obj.GetObjectKind().GroupVersionKind())
+}
+
+// Returns a string that is the form <namespace>:GroupVersionKind
+func NamespaceGroupVersionKindFromObject(obj *unstructured.Unstructured) string {
+	if obj == nil {
+		return ":"
+	}
+	return fmt.Sprintf("%s:%s", obj.GetNamespace(), obj.GetObjectKind().GroupVersionKind())
+}
```

Comment:
These two methods should be combined into one common method like this:
```
func NamespaceGroupVersionKindFromObject(namespace string, obj *unstructured.Unstructured) string {
	if obj == nil {
		return ":"
	}
	return fmt.Sprintf("%s:%s", namespace, obj.GetObjectKind().GroupVersionKind())
}
```
On second thought, why are they even different?  Just make the global unstructured object have the namespace name in it and use the second method.

Reply:
there is no "global" unstructured object. The difference between `GlobalGroupVersionKind` and `NamespaceGroupVersionKindFromObject` is that `GlobalGroupVersionKind` will return `_global:<GroupVersionKind of obj>` where `NamespaceGroupVersionKindFromObject` will use the namespace of the object instead (`<namespace of obj>:<GroupVersionKind of obj>`)

Reply:
> With having the global expressions separate I can treat every file the same. 

Why not? This should be *identical* to every other namespace in the `ConfigMap`. You should be able to read it in and run `yaml.Unmarshal` on it, and there you go.

Reply:
The global stuff is going away